### ISSUE TITLE
telegraf panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by landev2** [influxdata/telegraf#2378]
### Relevant telegraf.conf: 
agent.conf:

```
# Global tags can be specified here in key="value" format.
[global_tags]
  # dc = "us-east-1" # will tag all metrics with dc=us-east-1
  # rack = "1a"
  ## Environment variables can be used as tags, and throughout the config file
  # user = "$USER"


# Configuration for telegraf agent
[agent]
  ## Default data collection interval for all inputs
  interval = "60s"
  ## Rounds collection interval to 'interval'
  ## ie, if interval="10s" then always collect on :00, :10, :20, etc.
  round_interval = true

  ## Telegraf will send metrics to outputs in batches of at
  ## most metric_batch_size metrics.
  metric_batch_size = 1000
  ## For failed writes, telegraf will cache metric_buffer_limit metrics for each
  ## output, and will flush this buffer on a successful write. Oldest metrics
  ## are dropped first when this buffer fills.
  metric_buffer_limit = 10000

  ## Collection jitter is used to jitter the collection by a random amount.
  ## Each plugin will sleep for a random time within jitter before collecting.
  ## This can be used to avoid many plugins querying things like sysfs at the
  ## same time, which can have a measurable effect on the system.
  collection_jitter = "7s"

  ## Default flushing interval for all outputs. You shouldn't set this below
  ## interval. Maximum flush_interval will be flush_interval + flush_jitter
  flush_interval = "70s"
  ## Jitter the flush interval by a random amount. This is primarily to avoid
  ## large write spikes for users running a large number of telegraf instances.
  ## ie, a jitter of 5s and interval 10s means flushes will happen every 10-15s
  flush_jitter = "0s"

  ## By default, precision will be set to the same timestamp order as the
  ## collection interval, with the maximum being 1s.
  ## Precision will NOT be used for service inputs, such as logparser and statsd.
  ## Valid values are "ns", "us" (or "µs"), "ms", "s".
  precision = ""
  ## Run telegraf in debug mode
  debug = false
  ## Run telegraf in quiet mode
  quiet = true
  ## Override default hostname, if empty use os.Hostname()
  hostname = ""
  ## If set to true, do no set the "host" tag in the telegraf agent.
  omit_hostname = false
```
### Steps to reproduce:

1.  Let agent run... fails intermittently (dies) on various systems.

### Additional info: Various systems have had telegraf die since installing new version.  The service can be restarted... continues to run on some systems, dies again on others.

[Include gist of relevant config, logs, etc.]

cat /var/log/telegraf/telegraf.log
2017-02-07T11:38:46Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:39:49Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:40:59Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:42:06Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:43:10Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:44:15Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:45:23Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:46:32Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:47:41Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:48:45Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:49:50Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:50:56Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:52:03Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:53:12Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:54:17Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:55:22Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:56:27Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:57:30Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:58:37Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T11:59:44Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:00:51Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:01:59Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:03:07Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:04:14Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:05:20Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:06:26Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:07:31Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:08:36Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:09:41Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:10:51Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:11:55Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:13:01Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:14:07Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:15:17Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:16:24Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:17:33Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:18:40Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:19:49Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:20:56Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:22:01Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:23:08Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:24:17Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:25:26Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:26:31Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:27:38Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:28:43Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:29:51Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:31:01Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:32:07Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:33:11Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:34:19Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:35:27Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:36:32Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:37:36Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:38:44Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:39:51Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:40:56Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:41:59Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:43:05Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:44:08Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:45:13Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:46:18Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:47:26Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:48:33Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:49:38Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:50:42Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:51:50Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:52:59Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:54:07Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:55:12Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:56:18Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:57:25Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:58:34Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T12:59:40Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:00:46Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:01:50Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:02:54Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:04:01Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:05:06Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:06:13Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:07:18Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:08:21Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:09:27Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:10:33Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:11:41Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:12:50Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:13:57Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T13:15:01Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x654edb]

goroutine 13617 [running]:
panic(0xf2a2e0, 0xc420010090)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/influxdata/influxdb/client/v2.(*client).Write(0xc42015e0c0, 0x18fd660, 0xc420678cc0, 0x0, 0x0)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/influxdb/client/v2/client.go:354 +0xfb
github.com/influxdata/telegraf/plugins/outputs/influxdb.(*InfluxDB).Write(0xc42000a800, 0xc420025800, 0x3a, 0x3a, 0xc420707e58, 0x6b3911)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/outputs/influxdb/influxdb.go:202 +0x27b
github.com/influxdata/telegraf/internal/models.(*RunningOutput).write(0xc4201f2640, 0xc420025800, 0x3a, 0x3a, 0x3a, 0xc420707f28)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/internal/models/running_output.go:170 +0xa1
github.com/influxdata/telegraf/internal/models.(*RunningOutput).Write(0xc4201f2640, 0x11562e0, 0xc4201bc8d0)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/internal/models/running_output.go:154 +0x49c
github.com/influxdata/telegraf/agent.(*Agent).flush.func1(0xc4201bc8d0, 0xc4201f2640)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:232 +0x68
created by github.com/influxdata/telegraf/agent.(*Agent).flush
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:237 +0xb4
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x654edb]

goroutine 10040 [running]:
panic(0xf2a2e0, 0xc420010090)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/influxdata/influxdb/client/v2.(*client).Write(0xc4203800c0, 0x18fd660, 0xc4208ceba0, 0x0, 0x0)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/influxdb/client/v2/client.go:354 +0xfb
github.com/influxdata/telegraf/plugins/outputs/influxdb.(*InfluxDB).Write(0xc42000a800, 0xc42060a800, 0x3a, 0x3a, 0xc42045c658, 0x6b3911)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/outputs/influxdb/influxdb.go:202 +0x27b
github.com/influxdata/telegraf/internal/models.(*RunningOutput).write(0xc4201ec780, 0xc42060a800, 0x3a, 0x3a, 0x3a, 0xc42045c728)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/internal/models/running_output.go:170 +0xa1
github.com/influxdata/telegraf/internal/models.(*RunningOutput).Write(0xc4201ec780, 0x11562e0, 0xc4208fe940)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/internal/models/running_output.go:154 +0x49c
github.com/influxdata/telegraf/agent.(*Agent).flush.func1(0xc4208fe940, 0xc4201ec780)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:232 +0x68
created by github.com/influxdata/telegraf/agent.(*Agent).flush
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:237 +0xb4
2017-02-07T16:27:04Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:28:11Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:29:19Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:32:03Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:33:10Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:34:18Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:35:21Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:36:28Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:37:33Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:38:36Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:39:42Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:40:47Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:41:52Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:42:54Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:44:02Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)
2017-02-07T16:45:06Z E! ERROR: input [inputs.sysstat] took longer to collect than collection interval (1m0s)

**Comment by sparrc** [comment]
fixed in 1.2.1: https://github.com/influxdata/telegraf/issues/2317

**Comment by landev2** [comment]
Thank you Sir!
